src/Umbraco.Core/Services/DataTypeService.cs

```diff
@@ -355,50 +356,73 @@ private void ConvertMissingEditorsOfDataTypesToLabels(IEnumerable<IDataType> dat
+            return result.Status switch
             {
-                var moveEventInfo = new MoveEventInfo<IDataType>(toMove, toMove.Path, parentId);
+                DataTypeOperationStatus.Success => OperationResult.Attempt.Succeed(MoveOperationStatusType.Success, evtMsgs),
+                DataTypeOperationStatus.CancelledByNotification => OperationResult.Attempt.Fail(MoveOperationStatusType.FailedCancelledByEvent, evtMsgs),
+                DataTypeOperationStatus.ParentNotFound => OperationResult.Attempt.Fail(MoveOperationStatusType.FailedParentNotFound, evtMsgs),
+                _ =>  OperationResult.Attempt.Fail<MoveOperationStatusType>(MoveOperationStatusType.FailedNotAllowedByPath, evtMsgs, new InvalidOperationException($"Invalid operation status: {result.Status}")),
+            };
+        }
 
-                var movingDataTypeNotification = new DataTypeMovingNotification(moveEventInfo, evtMsgs);
-                if (scope.Notifications.PublishCancelable(movingDataTypeNotification))
-                {
-                    scope.Complete();
-                    return OperationResult.Attempt.Fail(MoveOperationStatusType.FailedCancelledByEvent, evtMsgs);
-                }
+        public async Task<Attempt<IDataType, DataTypeOperationStatus>> MoveAsync(IDataType toMove, Guid? containerKey, int userId = Constants.Security.SuperUserId)
+        {
+            EventMessages eventMessages = EventMessagesFactory.Get();
 
-                try
+            using (ICoreScope scope = ScopeProvider.CreateCoreScope())
+            {
+                EntityContainer? container = null;
+                var parentId = Constants.System.Root;
+                if (containerKey.HasValue && containerKey.Value != Guid.Empty)
                 {
-                    EntityContainer? container = null;
-                    if (parentId > 0)
+                    container = await _dataTypeContainerService.GetAsync(containerKey.Value);
+                    if (container is null)
                     {
-                        container = _dataTypeContainerRepository.Get(parentId);
-                        if (container == null)
-                        {
-                            throw new DataOperationException<MoveOperationStatusType>(MoveOperationStatusType.FailedParentNotFound); // causes rollback
-                        }
+                        return Attempt.FailWithStatus(DataTypeOperationStatus.ParentNotFound, toMove);
                     }
-                    moveInfo.AddRange(_dataTypeRepository.Move(toMove, container));
 
-                    scope.Notifications.Publish(new DataTypeMovedNotification(moveEventInfo, evtMsgs).WithStateFrom(movingDataTypeNotification));
+                    parentId = container.Id;
+                }
 
-                    scope.Complete();
+                if (toMove.ParentId == parentId)
+                {
+                    return Attempt.SucceedWithStatus(DataTypeOperationStatus.Success, toMove);
```

Comment:
Why are we returning success here when we haven't performed the move operation yet? This check used to result in Failed status:

https://github.com/umbraco/Umbraco-CMS/blob/v13/dev/src/Umbraco.Core/Services/DataTypeService.cs#L359-L362